### PR TITLE
feat: migrate `POST /api/users` endpoint to `POST /api/v1/users`

### LINKED ISSUE #145: [TASK] Migrate `POST /api/users` endpoint to `POST /api/v1/users`

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR adds a new endpoint `POST /api/v1/users` allowing the creation of a new user for API v1.

Changes of the new endpoint compared to the old one:
* `workspaces` attribute is not allowed when creating users with this endpoint. If necessary subsequent requests should be done to an endpoint that is still pending to be migrated. Specifically `POST /api/v1/workspaces/:workspace_id/users/:user_id`.
* In the same way that is happening with previous PRs the returned created user is not including `full_name` attribute.

Closes #145 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] Adding new tests and checking that the tests for API v0 are working as expected if using API v1 endpoint (except the ones creating workspaces along with the user).

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla_server/schemas/v1/users.py [hunk 1]
```diff
@@ -34,5 +38,13 @@ class Config:
         orm_mode = True
 
 
+class UserCreate(BaseModel):
+    first_name: constr(min_length=1, strip_whitespace=True)
+    last_name: Optional[constr(min_length=1, strip_whitespace=True)]
+    username: str = Field(regex=USER_USERNAME_REGEX, min_length=1)
```
Comment (frascuchon) [inline src/argilla_server/schemas/v1/users.py:44]: `regex` has been deprecated and replaced by `pattern` a long time ago.

```suggestion
    username: str = Field(pattern=USER_USERNAME_REGEX, min_length=1)
```

**Review by frascuchon — APPROVED** [review]

File: src/argilla_server/schemas/v1/users.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla_server/schemas/v1/users.py:44]: Not sure if it's our Pydantic version but it's not working using `pattern` parameter. It's not working even using `r` sigil to specify the regular expression like:

```python
USER_USERNAME_PATTERN = r"^(?!-|_)[A-za-z0-9-_]+$"
```

File: src/argilla_server/apis/v1/handlers/users.py [hunk 2]
```diff
@@ -49,6 +50,29 @@ async def list_users(
...
+    *,
+    db: AsyncSession = Depends(get_async_db),
+    user_create: UserCreate,
+    current_user: models.User = Security(auth.get_current_user),
+):
+    await authorize(current_user, UserPolicyV1.create)
+
+    user = await accounts.get_user_by_username(db, user_create.username)
+    if user is not None:
+        raise EntityAlreadyExistsError(name=user_create.username, type=User)
```
[the remark this answers is not in the document]
Comment (jfcalvo) [reply src/argilla_server/apis/v1/handlers/users.py:64]: Not sure if this should be moved to the validator but probably this should be inside the `accounts.create_user` function.

File: src/argilla_server/apis/v1/handlers/users.py [hunk 3]
```diff
@@ -49,6 +50,29 @@ async def list_users(
     return Users(items=users)
 
 
+@router.post("/users", response_model=User)
```
Comment (jfcalvo) [inline src/argilla_server/apis/v1/handlers/users.py:53]:
```suggestion
@router.post("/users", status_code=status.HTTP_201_CREATED, response_model=User)
```

File: tests/unit/api/v1/users/test_create_user.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+        )
+
+        assert response.status_code == 200
+
+        assert (await db.execute(select(func.count(User.id)))).scalar() == 2
+        user = (await db.execute(select(User).filter_by(username="username"))).scalar_one()
+
+        response_json = response.json()
+        assert response_json == {
+            "id": str(UUID(response_json["id"])),
```
Comment (jfcalvo) [inline tests/unit/api/v1/users/test_create_user.py:53]:
```suggestion
            "id": str(user.id),
```

File: tests/unit/api/v1/users/test_create_user.py [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+            "first_name": "First name",
+            "last_name": "Last name",
+            "username": "username",
+            "role": UserRole.annotator,
+            "api_key": user.api_key,
+            "inserted_at": datetime.fromisoformat(response_json["inserted_at"]).isoformat(),
+            "updated_at": datetime.fromisoformat(response_json["updated_at"]).isoformat(),
```
Comment (jfcalvo) [inline tests/unit/api/v1/users/test_create_user.py:60]:
```suggestion
            "inserted_at": user.inserted_at.isoformat(),
            "updated_at": user.updated_at.isoformat(),
```

File: src/argilla_server/apis/v1/handlers/users.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jfcalvo) [reply src/argilla_server/apis/v1/handlers/users.py]: I have moved the check for username uniqueness to `create_user` context function.

**Pull request merged**